testing/tests/test_e2e/test_secrets.py

```diff
@@ -661,17 +662,60 @@ def test_add_secret_with_metadata(secrets_context: Context[SecretsCharm], fields
         assert info['rotates'] is None
 
 
+@pytest.mark.parametrize('lookup_by', ['id', 'label'])
+@pytest.mark.parametrize(
+    'flow',
+    [
+        'content,label,description,expire,rotate',
+        'content,description,content,description',
+        'rotate,content,rotate,content,rotate',
+        'label,content,label,content',
+    ],
+)
+def test_set_secret(
+    secrets_context: Context[SecretsCharm], flow: str, lookup_by: Literal['id', 'label']
+):
+    secret = Secret({'some': 'content'}, owner='app', id='theid', label='thelabel')
+    state = State(leader=True, secrets={secret})
+    params = {'flow': flow, f'secret{lookup_by}': f'the{lookup_by}'}
+    state = secrets_context.run(secrets_context.on.action('set-secret-flow', params=params), state)
+    scenario_secret = state.get_secret(id='theid')
+    result = cast('Result', secrets_context.action_results)
+    assert 'after' in result
+    assert result['after']
+    info = result['after']['info']
+    assert info
+
+    common_assertions(scenario_secret, result)
+
+    counts = collections.Counter(flow.split(','))
+    if counts['content']:
+        assert result['after']['latest'] == {'val': str(counts['content'])}
+    if counts['label']:
+        assert info['label'] == f'label{counts["label"]}'
+    if counts['description']:
+        assert info['description'] == f'description{counts["description"]}'
+    if counts['expire']:
+        assert info['expires'] == datetime.datetime(2010 + counts['expire'], 1, 1, 0, 0)
+    if counts['rotate']:
+        rotation_values = ['sentinel', *SecretRotate.__members__.values()]
+        assert info['rotation'] == rotation_values[counts['rotate']]
+
+
 def common_assertions(scenario_secret: Secret | None, result: Result):
     if scenario_secret:
-        assert scenario_secret.owner == 'application'
+        # https://github.com/canonical/operator/issues/2125
+        assert scenario_secret.owner in ('app', 'application')
         assert not scenario_secret.remote_grants
 
         assert result.get('after')
         info = result['after']['info']
         # Verify that the unit and the scaffolding see the same data
-        assert scenario_secret.id == info['id']
+        #
+        # Scenario presents a secret with a full secret URI to the charm
+        # however, the id on the scenario Secret object is a plain id
+        assert scenario_secret.id.split('/')[-1] == info['id'].split('/')[-1]
         assert scenario_secret.label == info['label']
-        assert scenario_secret._tracked_revision == info['revision']
```

Comment:
Since this is being removed, as it's not always going to be the case now, should there be some `assert scenario_secret...` type assertions in the individual tests to replace it? At the moment we have tests in this common function that the Scenario object matches the data loaded in the charm code, and in the individual test checks that the data loaded in the charm code matches the initial state. Maybe that's all you're interested in, in which case you can just ignore this.